org.eclipse.xtext/src/org/eclipse/xtext/generator/URIBasedFileSystemAccess.xtend

```diff
@@ -93,9 +94,13 @@ class URIBasedFileSystemAccess extends AbstractFileSystemAccess2 {
 		if (isGenerateTraces && contents instanceof ITraceRegionProvider) {
 			var String traceFileName = traceFileNameProvider.getTraceFromJava(generatedFile)
 			val out = new ByteArrayOutputStream()
-			var AbstractTraceRegion traceRegion = (contents as ITraceRegionProvider).getTraceRegion()
-			traceRegionSerializer.writeTraceRegionTo(traceRegion, out)
-			generateFile(traceFileName, outputConfigName, new ByteArrayInputStream(out.toByteArray))
+			try {
+				var AbstractTraceRegion traceRegion = (contents as ITraceRegionProvider).getTraceRegion()
```

Comment:
Same as above: call `getTraceRegion()` first to avoid unnecessary computations.

Reply:
I agree - it would make sense to postpone `getTraceFromJava` and do it only if there is a traceRegion.